plugin/src/main/java/org/gradle/testretry/TestRetryPlugin.java

```diff
@@ -39,9 +39,16 @@
 
     @Override
     public void apply(Project project) {
+        if (pluginAlreadyApplied(project)) {
+            return;
+        }
+
         project.getTasks()
             .withType(Test.class)
             .configureEach(task -> configureTestTask(task, objectFactory, providerFactory));
     }
 
+    private static boolean pluginAlreadyApplied(Project project) {
+        return project.getPlugins().stream().anyMatch(plugin -> plugin.getClass().getName().equals(TestRetryPlugin.class.getName()));
```

Comment:
Why not using directly 
```suggestion
        return project.getPlugins().stream().anyMatch(plugin -> plugin.getClass().equals(TestRetryPlugin.class));
 ```

Reply:
is this Config Cache safe?

Reply:
@jonnybbb We're comparing the name because the class may be coming from 2 different classloaders. 

@paplorinc This is the strategy we use for the build scan plugin, so I believe it is config cache-safe.